### ISSUE TITLE
How close to a drop-in replacement?

### ISSUE THREAD

**Issue by binkley** [imagingbook/parboiled-modular#6]
*tl;dr* -- **what's the smallest change for a project to migrate from Parboiled 1.4.1?**

I saw your helpful comment in:
https://github.com/sirthias/parboiled/issues/147#issuecomment-1420949182

I have a project stuck on JDK 11 because of the issues you addressed with modularity:
https://github.com/binkley/kotlin-dice

So I tried the simplest thing I could and left the project using JDK 11:
```diff
-        <parboiled.version>1.4.1</parboiled.version>
+        <parboiled-modular.version>1.1.0</parboiled-modular.version>

         <dependency>
-            <groupId>org.parboiled</groupId>
+            <groupId>com.imagingbook</groupId>
             <artifactId>parboiled-java</artifactId>
-            <version>${parboiled.version}</version>
+            <version>${parboiled-modular.version}</version>
         </dependency>
```

However, I'm getting parse errors with unresolved references to many Parboiled methods, eg, `pop` and `push`, `AnyOf`, `ZeroOfMore`, et al. Now it could be something related to that the code is in Kotlin, however, this is not the kind of problem I've seen when Kotlin code calls into Java libraries.

**Comment by imagingbook** [comment]
Could it be that you did not update the `import` statements for classes `BaseActions`, `BaseParser` and `Parboiled` (as described in [README.md](https://github.com/imagingbook/parboiled-modular/blob/master/README.md#features))? These are the only ones that had to move to a new package.  

For example, in file `DiceParser.kt` it should read `import org.parboiled.parser.BaseParser` instead of `import org.parboiled.BaseParser`, etc.

**Comment by binkley** [comment]
@imagingbook I am not patient, and you are very patient! Thank you for pointing out what I should have investigated in the first place -- I read the compile errors, and "import" didn't leap to mind.

Migrating is more comprehensive than I expected, but have a better idea what to focus on. For example, migrating to JDK 17 from 11 is now mandatory (class file version changes), but this has been a long-time goal for the `kotlin-dice` project. Thank you for making that possible.

**Comment by imagingbook** [comment]
Great! As far as `parboiled-modular` is concerned, there is no need to step to JDK 17. It could be compiled for target level 11 just as well, I suppose ...
